pkg/util/oom/oom_linux.go

```diff
@@ -78,12 +75,22 @@ func applyOOMScoreAdj(pid int, oomScoreAdj int) error {
 	value := strconv.Itoa(oomScoreAdj)
 	var err error
 	for i := 0; i < maxTries; i++ {
-		if err = ioutil.WriteFile(oomScoreAdjPath, []byte(value), 0700); err != nil {
+		f, err := os.Open(oomScoreAdjPath)
+		if err != nil {
+			if syscallNotExists(err) {
+				return os.ErrNotExist
```

Comment:
This logic has grown organically complex. Your suggestion makes sense. Although, this whole logic will be removed once we can drop support for docker versions < v1.10.